### PR TITLE
Keras Tuner - Fix model retrain parameters

### PR DESCRIPTION
**Pull request by yiannisEfth**
Resolves: https://github.com/tensorflow/tensorflow/issues/47725

### CODE REVIEW

**Comment by yiannisEfth** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Fdocs%2Fpull%2F1844) for more info**.

@googlebot I signed it!

**Comment by yiannisEfth** [comment]
Resolves: https://github.com/tensorflow/tensorflow/issues/47848

**Comment by lamberta** [comment]
May need to rebase to the latest

**Comment by yiannisEfth** [comment]
@lamberta Done

**Comment by 8bitmp3** [comment]
Hi @yiannisEfth @jvishnuvardhan. I'm just following the tutorial myself and I want to make sure that adding `validation_split=0.2` in the last loop is what is currently missing, as suggested by @yiannisEfth 's PR.

1. When you run the hyperparam search, you specify the `validation_split` (at 20%) and save the optimal hyperparams in `best_hps`:

```py
tuner.search(img_train, label_train, epochs=50, validation_split=0.2, callbacks=[stop_early])

# Get the optimal hyperparameters
best_hps=tuner.get_best_hyperparameters(num_trials=1)[0]
```

2. Then, to find the optimal number of epochs with `best_hps`, you train for 50 epochs with `validation_split`, again, at 20%:

```py
model = tuner.hypermodel.build(best_hps)
history = model.fit(img_train, label_train, epochs=50, validation_split=0.2)
```

3. Finally, you retrain the model with the optimal number of epochs with `validation_split=0.2`, as proposed by @yiannisEfth:

```diff
# Retrain the model
- hypermodel.fit(img_train, label_train, epochs=best_epoch)
+ hypermodel.fit(img_train, label_train, epochs=best_epoch, validation_split=0.2)
```

Did I get this right? Cheers 👋

**Comment by yiannisEfth** [comment]
@8bitmp3 Yes, you use these `val_loss `and `val_accuracy ` metrics derived from the `validation_split` after each epoch to get a sense of when your model has hit the best performance. 
We can be confident that we should get the exact same `val_accuracy` and `val_loss` numbers when we repeat the process from step 2 in step 3 (if we've set seeds for replicability of course), since we use the exact same hyperparameters. 
Keep in mind though, these can still vary every time you fit the model, especially if you use a GPU back-end, due to sophisticated libraries.

**Review by MarkDaoust — DISMISSED** [review]

**Comment by MarkDaoust** [comment]
> if we've set seeds for replicability of course

Fit's `validation_split` argument takes the data from the end ([from here](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit)):

> The validation data is selected from the last samples in the x and y data provided, before shuffling

**Review by lamberta — APPROVED** [review]
